apis/openstack-swift/src/test/java/org/jclouds/openstack/swift/v1/features/DynamicLargeObjectApiLiveTest.java

```diff
@@ -0,0 +1,175 @@
+
+      SwiftObject bigObject = getApi().getObjectApi(regionId, defaultContainerName).get(objectName);
+      assertThat(bigObject.getETag().equals(etagOfEtags));
+      assertThat(bigObject.getPayload().getContentMetadata().getContentLength().equals(Long.valueOf(total_size)));
+      assertThat(bigObject.getMetadata().equals(ImmutableMap.of("myfoo", "Bar")));
+      assertThat(getApi().getContainerApi(regionId).get(defaultContainerName).getObjectCount().equals(Long.valueOf(3)));
+   }
+
+   protected void assertReplaceManifest(String regionId, String containerName, String name) {
+      ObjectApi objectApi = getApi().getObjectApi(regionId, containerName);
+
+      String etag1s = objectApi.put(name + "/1", newByteSourcePayload(megOf1s));
+      awaitConsistency();
+      assertMegabyteAndETagMatches(regionId, containerName, name + "/1", etag1s);
+
+      String etag2s = objectApi.put(name + "/2", newByteSourcePayload(megOf2s));
+      awaitConsistency();
+      assertMegabyteAndETagMatches(regionId, containerName, name + "/2", etag2s);
+
+      List<Segment> segments = ImmutableList.<Segment> builder()
+            .add(Segment.builder().path(format("%s/%s/1", containerName, name)).etag(etag1s).sizeBytes(1024 * 1024)
+                  .build())
+            .add(Segment.builder().path(format("%s/%s/2", containerName, name)).etag(etag2s).sizeBytes(1024 * 1024)
+                  .build())
+            .build();
+
+      awaitConsistency();
+      String etagOfEtags = getApi().getDynamicLargeObjectApi(regionId, containerName).replaceManifest(name, segments,
+            ImmutableMap.of("myfoo", "Bar"), ImmutableMap.of("header1", "value1"));
+
+      assertNotNull(etagOfEtags);
+
+      awaitConsistency();
+
+      SwiftObject bigObject = getApi().getObjectApi(regionId, containerName).get(name);
+      assertThat(bigObject.getETag().equals(etagOfEtags));
+      assertThat(bigObject.getPayload().getContentMetadata().getContentLength().equals(Long.valueOf(2 * 1024 * 1024)));
+      assertThat(bigObject.getMetadata().equals(ImmutableMap.of("myfoo", "Bar")));
+
+      // segments are visible
+      assertThat(getApi().getContainerApi(regionId).get(containerName).getObjectCount().equals(Long.valueOf(3)));
+   }
+
+   protected void assertMegabyteAndETagMatches(String regionId, String containerName, String name, String etag1s) {
+      SwiftObject object1s = getApi().getObjectApi(regionId, containerName).get(name);
+      assertThat(object1s.getETag().equals(etag1s));
+      assertThat(object1s.getPayload().getContentMetadata().getContentLength().equals(Long.valueOf(1024 * 1024)));
+   }
+
+   protected void deleteAllObjectsInContainerDLO(String regionId, final String containerName) {
+       ObjectList objects = getApi().getObjectApi(regionId, containerName).list(new ListContainerOptions());
+      if (objects == null) {
+         return;
+      }
+      List<String> pathsToDelete = Lists.transform(objects, new Function<SwiftObject, String>() {
```

Comment:
Before looping, the object by their name for delete, we need list of object in container.